cli/tests/pcluster/aws/dummy_aws_api.py

```diff
@@ -87,11 +88,11 @@ def instance_storage_supported(self):
 
 
 class _DummyAWSApi(AWSApi):
-    def __init__(self):
+    def __init__(self, non_happy=False):
```

Comment:
I have a strong concern about this approach.
A user of this constructor cannot understand the implications of setting `non_happy`:
1. setting `non_happy` at the constructor level would make me think that it would imply non happy flows in every client, instead it does it only on FSx;
2. what does `non_happy` mean? which calls are affected by this settings?

Reply:
I could rename the parameter to `test_non_happy_flow` for clarity. This can be applied to any test needing a mock for a non-happy flow. Also, as it is a keyword argument it has no impact on existing code and can facilitate any future testing of non-happy flows.

Reply:
I think we should avoid adding this parameter. What if we want to inject failure in N API calls? With the approach you're proposing we should add N parameters to the `DummyAwsApi` constructor and it would not scale.

what about something like the following?

```
def your_unit_test(mocker):
    dummy_api = _DummyAWSApi()
    dummy_api._fsx.make_describe_volumes_fail_with_error(True, error=UnauthorizedException)
    mocker.patch("pcluster.aws.aws_api.AWSApi.instance", return_value=dummy_api)

   ... your unit tests logic ...
```

Reply:
I like that approach